views/aggPool.go

```diff
@@ -50,17 +53,30 @@ func (v *Views) QueryPoolCandles(chainId types.ChainId, base types.EthAddress, q
 	startTime := endTime - timeRange.N*timeRange.Period
 
 	if timeRange.StartTime != nil {
-		startTime = *timeRange.StartTime
-		endTime = startTime + timeRange.N*timeRange.Period
-	}
 
+		if(uniswapCandles){
+            endTime = *timeRange.StartTime + timeRange.Period
+            startTime = *timeRange.StartTime - timeRange.N *timeRange.Period
+        }else {
+            startTime = *timeRange.StartTime
+            endTime = startTime + timeRange.N*timeRange.Period
+        }
+	}
 	open, series := v.Cache.RetrievePoolAccumSeries(loc, startTime, endTime)
 
 	builder := model.NewCandleBuilder(startTime, timeRange.Period, open)
 	for _, accum := range series {
 		builder.Increment(accum)
 	}
-	return builder.Close(endTime)
+	 candles := builder.Close(endTime)
+
+	 if(uniswapCandles){
```

Comment:
is there technical debt here where we could just store them backwards?

Reply:
So the normal GCGO needs to send them forwards, the other one needs to store them backward. 

We could do it with one forward sort, but then we'd have to adjust how the candle builder looks for end ranges. 

It takes very little time to resort, but yeah we could technically fix it so it's not needed.